src/LndClient/Data/AddInvoice.hs

```diff
@@ -10,17 +10,15 @@ where
 import Crypto.Hash.SHA256 (hash)
 import Data.Aeson (FromJSON (..), ToJSON (..))
 import Data.ByteString.Base64 (encode)
-import Data.Text (Text)
 import Data.Text.Encoding (decodeUtf8, encodeUtf8)
 import GHC.Generics (Generic)
+import LndClient.Data.Invoice (Invoice (..))
 import LndClient.Data.Newtypes
 import LndClient.Utils (stdParseJSON, stdToJSON)
 
-data AddInvoiceRequest
+newtype AddInvoiceRequest
```

Comment:
If we decide to go with custom AddInvoiceRequest parameters instead of generic Invoice (because of optional parameters mess) - we might not need this "result" newtype, because there is already "ResultWrapper" generic type which you can apply inside method implementation in "RPC" module (how it's done for "subscribeInvoices")

Reply:
Done, using ResultWrapper + AddInvoiceRequest, with specific attributes for adding only (instead if Invoice)